src/framework/JetScapeWriterFinalStateStream.cc

```diff
@@ -142,17 +151,26 @@ template <class T> void JetScapeWriterFinalStateStream<T>::Init() {
     particleStatusToSkip = detail::stringToVector(s);
   }
   if (GetActive()) {
+    // We need the header version to determine how to write.
+    // NOTE: Don't require the version. If not specified, defaults to v2.
+    int result = GetXMLElementInt({"final_state_writer_header_version"}, false);
```

Comment:
We already discussed and I know it's in progress, so just noting here so we don't forget - we'll merge this option into the other writer options